### ISSUE TITLE
Inspector service not working due to null location - detailed information - 1st time only.

### ISSUE THREAD

**Issue by matanshukry** [flutter/flutter#84423]
## Steps to Reproduce

1. Run my flutter app
2. Open the Flutter Inspector in Android Studio
3. No errors and an error shows up
4. (Optional) Keep hitting the refresh button to repeat

**Expected results:**
List of widgets available in the inspector widget + no error

**Actual results:**

```
======== Exception caught by Flutter framework =====================================================
The following _TypeError was thrown during a service extension callback for "ext.flutter.inspector.getRootWidgetSummaryTree":
type 'Null' is not a subtype of type 'String' of 'function result'

When the exception was thrown, this was the stack: 
#0      _Location.file (package:flutter/src/widgets/widget_inspector.dart)
#1      _Location.toJsonMap (package:flutter/src/widgets/widget_inspector.dart:2844:15)
#2      _Location.toJsonMap.<anonymous closure> (package:flutter/src/widgets/widget_inspector.dart:2853:42)
#3      MappedListIterable.elementAt (dart:_internal/iterable.dart:412:31)
#4      ListIterator.moveNext (dart:_internal/iterable.dart:341:26)
...
====================================================================================================
```

**More information:**
I debugged the framework code a bit to find the issues, and ended up at `widget_inspector.dart` file at `_Location::toJsonMap`, where it's calling `.toJsonMap()` on each item in `parameterLocations`. As you can see from the picture below, the `File` member is null, although it's not Optional/nullable, so not sure how. That what seems to be causing the issues.

Also, this seems to be working once I restarted the app. Looks like this is happening only after clean/the 1st time.
If the flutter team is fine with just saying one must restart the app after starting the inspector for the 1st time, perhaps the IDE should support this flow/behavior.

p.s.
I'm using `GoogleMap` widget from `google_maps_flutter` package. It's the only thing kind of "special" in my app that I can think may cause this.

![image](https://user-images.githubusercontent.com/5613042/121662038-8b4a7e00-caad-11eb-9141-df56baed483a.png)

<details>
  <summary>Logs</summary>

```
Analyzing v1...
No issues found! (ran in 5.2s)
```

```
[√] Flutter (Channel stable, 2.2.1, on Microsoft Windows [Version 10.0.19042.1052], locale en-US)
    • Flutter version 2.2.1 at D:\dev\Flutter
    • Framework revision 02c026b03c (2 weeks ago), 2021-05-27 12:24:44 -0700
    • Engine revision 0fdb562ac8
    • Dart version 2.13.1

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at D:\dev\Android\Sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = D:\dev\Android\Sdk
    • ANDROID_SDK_ROOT = D:\dev\Android\Sdk
    • Java binary at: D:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at D:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)

[√] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at D:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.3.2
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] Connected device (3 available)
    • sdk gphone x86 arm (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 91.0.4472.101
    • Edge (web)                  • edge          • web-javascript • Microsoft Edge 91.0.864.41

• No issues found!
```

</details>

Edit: while restarts seem to solve it, it seems to be happening more and more, and it only fixes it for a very short duration - even without any hot reloads in.

**Comment by TahaTesser** [comment]
Hi @matanshukry 
Thanks for filing the issue. 

I just tried the official maps sample with [google maps in a column](https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/google_maps_flutter/example/lib/scrolling_map.dart), couldn't reproduce this issue

<details> 
<summary>preview</summary> 

![image](https://user-images.githubusercontent.com/48603081/121866734-e37cbc80-cd07-11eb-8830-a59261c9b688.png)

</details>

Make sure you're using the latest plugin in Android Studio and try to reproduce the issue using terminal 

Execute `flutter run -v` and open devtools link at the bottom
example"
```
[   +8 ms] Connected to _flutterView/0x74394ccd40.
[   +4 ms] Flutter run key commands.
[   +1 ms] r Hot reload. 🔥🔥🔥
[   +1 ms] R Hot restart.
[        ] h Repeat this help message.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] 💪 Running with sound null safety 💪
[   +1 ms] An Observatory debugger and profiler on IN2011 is available at: http://127.0.0.1:57654/mVQFmCoKosU=/
[ +803 ms] The Flutter DevTools debugger and profiler on IN2011 is available at:
                 http://127.0.0.1:9104?uri=http%3A%2F%2F127.0.0.1%3A57654%2FmVQFmCoKosU%3D%2F
```

If the issue is specific to code sample, please provide reproducible code sample

**Comment by matanshukry** [comment]
@TahaTesser Checked again with the IDE since my app changed quickly over the past few days, and it seems to be gone.

I believe this is due to the OS image architecture, which caused multiple issues; specifically, I'm now using x86_64 image while before i was using x86 image. FYI, the x86 image is also causing a constant messages of "system UI isn't responding".

Not sure why exactly it's that way, but it might be worth adding it to the documentation. I'm closing this for now, feel free to open it as an issue about the docs though.
